test/COBRARequisitesFullfilled.m

```diff
@@ -0,0 +1,276 @@
+%                   * 'Toolboxes'      - Names of toolboxes (the license
+%                                        feature name) (Default: {})
+%                   * 'ReqSolvers'     - Names of all solvers which MUST be
+%                                        available. If not empty, the resulting
+%                                        solvers struct will contain cell arrays (Default: {})
+%                   * 'UseIfAvailable' - Names of solvers which should be
+%                                        used if they are available. If not empty, the resulting
+%                                        solvers struct will contain cell arrays (will
+%                                        not throw an error if not).
+%                   * 'NeedsLP'        - Whether a LP solver is required.
+%                                       (Default = false);
+%                   * 'NeedsMILP'      - Whether a MILP solver is required.
+%                                       (Default = false);
+%                   * 'NeedsQP'        - Whether a QP solver is required.
+%                                       (Default = false);
+%                   * 'NeedsMIQP'      - Whether a MIQP solver is required.
+%                                       (Default = false);
+%                   * 'NeedsNLP'       - Whether a NLP solver is required.
+%                                       (Default = false);
+%                   * 'NeedsUnix'      - Whether the test only works on a
+%                                        Unix system (mac or linux)
+%                                        (Default = false);
+%                   * 'NeedsWindows'   - Whether the test only works on a Windows system
+%                                        (Default = false);
+%                   * 'NeedsMac'       - Whether the test only works on a Mac system
+%                                        (Default = false);
+%                   * 'NeedsLinux'     - Whether the test only works on a Linux system
+%                                        (Default = false);
+%
+% OUTPUTS:
+%
+%    solversToUse:         A struct with one field per solver type listing
+%                          the solvers To use for that type of problem.
+%                          If neither the useIfAvailable nor the reqSolvers
+%                          paramter is provided, only at most one solver
+%                          per type will be returned (an empty string if
+%                          none is present). If either is provided, the
+%                          returned struct will contain cell arrays of
+%                          solver names.
+
+%Do some precomputation.
+global CBT_MISSING_REQUIREMENTS;
+global OPT_PROB_TYPES
+persistent availableSolvers
+
+%Some Matlab Toolboxes currently in use in the COBRA Toolbox.
+%This might have to be extended in the future.
+toolboxInfo = struct('statistics_toolbox',{{'Statistics and Machine Learning Toolbox','Statistics Toolbox'}},...
```

Comment:
It might be good to put this list to the doc (e.g., [here](https://opencobra.github.io/cobratoolbox/stable/installation.html))